.github/workflows/backend-service-cd.yml

```diff
@@ -0,0 +1,33 @@
+name: backend-service Continuous deployment
+
+on:
+  push:
+    branches:
+      # - 3-endpoint-and-healthcheck
+  workflow_dispatch:
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v3
+
+    - name: Rebuild and deploy backend container
+      env:
+        SSH_PRIVATE_KEY: ${{ secrets.SSH_PRIVATE_KEY }}
```

Comment:
I'm open to adding a `dev` script to the package.json but I'm having weird issues getting nodemon to play nice with typescript.

tsc-node-dev also screams at my saying it's leaking memory.

vitest gives hot reloads when you're trying to pass tests. That's usually the place I actually want the hot reload on backend stuff. I'm happy enough for now but let me know.

Reply:
Yeah for sure I can show you what's going on with that workflow if you want.

It's really just a case of passing the SSH key and server details rather than AWS/other cloud logins

Reply:
I'm happy enough with this deploy approach for now 👍